### PR TITLE
add canary release for kruise rollout

### PR DESCRIPTION
**Pull request by zmberg**
Signed-off-by: liheng.zms <liheng.zms@alibaba-inc.com>

 How to implement Canary Release using Kruise Rollout DOC.

### CODE REVIEW

**Review by wonderflow — COMMENTED** [review]
can we reduce or don't use kubectl in the docs?

File: docs/end-user/traits/canary-release.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+1. Install Kruise Rollout
+```shell
+% vela addon enable kruise-rollout
+enable addon by local dir: addons/kruise-rollout
+I0531 17:23:35.526011   81152 apply.go:107] "creating object" name="kruise-rollout" resource="core.oam.dev/v1beta1, Kind=TraitDefinition"
+I0531 17:23:35.688687   81152 apply.go:107] "creating object" name="addon-secret-kruise-rollout" resource="/v1, Kind=Secret"
+⠸ Waiting addon application running. It is now in phase: runningWorkflow (timeout 0/600 seconds)...
+Addon: kruise-rollout enabled Successfully.
+
+% kubectl get pods -n kruise-rollout
```
Comment (wonderflow) [inline docs/end-user/traits/canary-release.md:21]: we can use `vela addon status` to check the installation

File: docs/end-user/traits/canary-release.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+NAME                                                 READY   STATUS    RESTARTS   AGE
+kruise-rollout-controller-manager-696bdb76f8-2vqp8   1/1     Running   0          7m21s
+kruise-rollout-controller-manager-696bdb76f8-mbplq   1/1     Running   0          7m21s
+```
+
+2. Install K8S Nginx Ingress
+```shell
+helm upgrade --install ingress-nginx ingress-nginx --repo https://kubernetes.github.io/ingress-nginx --namespace ingress-nginx
```
Comment (wonderflow) [inline docs/end-user/traits/canary-release.md:29]: can we make it as another addon? @wangyikewxgm

File: docs/end-user/traits/canary-release.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+echoserver   5/5     5            5           130m
+
+% kubectl get service
+NAME         TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
+echoserver   ClusterIP   10.96.17.100   none        8080/TCP   19s
+
+% kubectl get ingress
+NAME         CLASS    HOSTS                    ADDRESS     PORTS   AGE
+echoserver   none   echoserver.example.com   localhost   80      26s
+```
```
Comment (wonderflow) [inline docs/end-user/traits/canary-release.md:92]: use `vela status` instead

File: docs/end-user/traits/canary-release.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+# when rollout State = Complated, indicates full release completed, and currently the application is ready
+% kubectl get rollouts
+NAME         STATUS    CANARY_STEP   CANARY_STATE   MESSAGE                                   AGE
+echoserver   Healthy   1             Completed      Rollout has been completed, and succeed   119m
+
+% kubectl get applications
+NAME         COMPONENT    TYPE         PHASE     HEALTHY   STATUS      AGE
+echoserver   echoserver   webservice   running   true      Ready:5/5   120m
+```
```
Comment (wonderflow) [inline docs/end-user/traits/canary-release.md:190]: do we have any way around vela cli , such as `vela workflow reusme` ?

**Comment by wangyikewxgm** [comment]
contained by: #723

**Pull request closed without merging**